core/kazoo_bindings/src/kazoo_bindings.erl

```diff
@@ -710,7 +713,9 @@ map_processor_fold(#kz_binding{binding_parts=BParts
                   ,_Routing
                   ,RoutingParts
                   ) ->
-    case matches(BParts, RoutingParts) of
+    case matches(BParts, RoutingParts)
+        orelse matches(RoutingParts, BParts)
```

Comment:
The first argument to matches/2 is the binding key; the second argument is the routing key. Binding keys allow wildcards; routing keys are concrete and do not have wildcards. Did you look at the [tests](https://github.com/2600hz/kazoo/blob/master/core/kazoo_bindings/test/kazoo_bindings_test.erl)?

I don't understand what you're trying to accomplish. Give me something concrete here. As-is, your changes have introduced ambiguity and muddied the water for how routing keys are matched to binding keys.